src/lv_ex_mbox/lv_ex_mbox_2.py

```diff
@@ -0,0 +1,86 @@
+welcome_info = "Welcome to the modal message box demo!\nPress the button to display a message box."
+in_msg_info = "Notice that you cannot touch the button again while the message box is open."
+
+class Modal(lv.mbox):
+    """mbox with semi-transparent background"""
+    def __init__(self, parent, *args, **kwargs):
+        # Create a full-screen background
+        modal_style = lv.style_t()
+        lv.style_copy(modal_style, lv.style_plain_color)
+        # Set the background's style
+        modal_style.body.main_color = modal_style.body.grad_color = lv.color_make(0,0,0)
+        modal_style.body.opa = lv.OPA._50
+        
+        # Create a base object for the modal background
+        self.bg = lv.obj(parent)
+        self.bg.set_style(modal_style)
+        self.bg.set_pos(0, 0)
+        self.bg.set_size(parent.get_width(), parent.get_height())
+        self.bg.set_opa_scale_enable(True)  # Enable opacity scaling for the animation
+
+        super().__init__(self.bg, *args, **kwargs)
+        self.align(None, lv.ALIGN.CENTER, 0, 0)
+
+        # Fade the message box in with an animation
+        a = lv.anim_t()
+        lv.anim_init(a)
+        lv.anim_set_time(a, 500, 0)
+        lv.anim_set_values(a, lv.OPA.TRANSP, lv.OPA.COVER)
+        lv.anim_set_exec_cb(a, self.bg, lv.obj.set_opa_scale)
+        lv.anim_create(a)
+        super().set_event_cb(self.default_callback)
+
+    def set_event_cb(self, callback):
+        self.callback = callback
+
+    def get_event_cb(self):
+        return self.callback
+
+    def default_callback(self, obj, evt):
+        if evt == lv.EVENT.DELETE:# and obj == self:
+            # Delete the parent modal background
+            self.get_parent().del_async()
+        elif evt == lv.EVENT.VALUE_CHANGED:
+            # A button was clicked
+            self.start_auto_close(0)
+        # Call user-defined callback
+        if self.callback is not None:
+            self.callback(obj, evt)
+
+def mbox_event_cb(obj, evt):
+    if evt == lv.EVENT.DELETE:
+        info.set_text(welcome_info)
+
+def btn_event_cb(btn, evt):
+    if evt == lv.EVENT.CLICKED:
+
+        btns2 = ["Ok", "Cancel", ""]
+
+        # Create the message box as a child of the modal background
+        mbox = Modal(lv.scr_act())
+        mbox.add_btns(btns2)
```

Comment:
Is `add_btns` working for you on `dev-6.1`? I'm getting: `SyntaxError: Cannot convert 'list' to pointer!`

`add_btns` [prototype](https://github.com/littlevgl/lvgl/blob/b4b6a472fb54c4f11b848fd3ee3910b65d6fee79/src/lv_objx/lv_mbox.h#L97) is:
```c
void lv_mbox_add_btns(lv_obj_t * mbox, const char ** btn_mapaction);
```

@kisvegabor - probably this too needs to be changed to an array of char* instead of a pointer to char*, for the binding to recognize this as an array and not a pointer.

Reply:
Yes, it works for me both on unixport and in the javascript simulator... Using `dev-6.1`.
[online simulator](https://littlevgl.github.io/sim/micropython/ports/javascript/lvgl_editor.html?script_direct=786e686a3a4ce922fe9aaaa3ea9a4b13315a7cfa)

Reply:
> Yes, it works for me both on unixport and in the javascript simulator... Using `dev-6.1`.
> [online simulator](https://littlevgl.github.io/sim/micropython/ports/javascript/lvgl_editor.html?script_direct=786e686a3a4ce922fe9aaaa3ea9a4b13315a7cfa)

The online simulator on https://littlevgl.github.io/sim/ is not up to date, AFAIK.  
But if you are saying this is working for you on the unix simulator with up-to-date dev-6.1... Maybe some local problem on my side. I'll recheck.

Reply:
Checked again - same problem. I think we are not aligned to the same lvgl vesion. Did you update submodules?  
Anyway, I've submitted a PR for fixing this on lvgl: https://github.com/littlevgl/lvgl/pull/1279